tstime/tstime.go

```diff
@@ -105,10 +104,15 @@ func Parse3339(s string) (time.Time, error) {
 }
 
 func parseInt(s string, dst *int) bool {
-	n, err := strconv.ParseInt(s, 10, 0)
-	if err != nil || n < 0 {
+	if len(s) == 0 || len(s) > len("999999999") {
 		return false
 	}
-	*dst = int(n)
+	*dst = 0
+	for i := range s {
```

Comment:
Also, you don't want to do a UTF-8-parsing range over string here. You want a normal for loop:

```
   for i := 0; i < len(s); i++ {
```